src/base_object-inl.h

```diff
@@ -201,7 +201,7 @@ void BaseObject::decrease_refcount() {
   unsigned int new_refcount = --metadata->strong_ptr_count;
   if (new_refcount == 0) {
     if (metadata->is_detached) {
-      delete this;
+      OnGCCollect();
```

Comment:
will this cause behavioral change to other subclasses of `BaseObject` ? (for example those who do not have custom implementations of `OnGCCollect` and expects content to be deleted through the old code)

Reply:
There are no subclasses other than `HandleWrap` which have custom `OnGCCollect` behavior, so, no.